sdk/core/ts-http-runtime/src/policies/auth/apiKeyAuthenticationPolicy.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { ApiKeyCredential } from "../../auth/credentials.js";
+import type { AuthScheme } from "../../auth/schemes.js";
+import type { PipelineRequest, PipelineResponse, SendRequest } from "../../interfaces.js";
+import type { PipelinePolicy } from "../../pipeline.js";
+import { ensureSecureConnection } from "./checkInsecureConnection.js";
+
+/**
+ * Name of the API Key Authentication Policy
+ */
+export const apiKeyAuthenticationPolicyName = "apiKeyAuthenticationPolicy";
+
+/**
+ * Options for configuring the API key authentication policy
+ */
+export interface ApiKeyAuthenticationPolicyOptions {
+  /**
+   * The credential used to authenticate requests
+   */
+  credential: ApiKeyCredential;
+  /**
+   * Optional authentication schemes to use. If not provided, schemes from the request will be used.
+   */
+  authSchemes?: AuthScheme[];
+  /**
+   * Allows for connecting to HTTP endpoints instead of enforcing HTTPS.
+   * CAUTION: Never use this option in production.
+   */
+  allowInsecureConnection?: boolean;
+}
+
+/**
+ * Gets a pipeline policy that adds API key authentication to requests
+ */
+export function apiKeyAuthenticationPolicy(
+  options: ApiKeyAuthenticationPolicyOptions,
+): PipelinePolicy {
+  return {
+    name: apiKeyAuthenticationPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      // Ensure allowInsecureConnection is explicitly set when sending request to non-https URLs
+      ensureSecureConnection(request, options);
+
+      const scheme = (request.authSchemes ?? options.authSchemes)?.find((x) => x.kind === "apiKey");
+
+      // Skip adding authentication header if no API key authentication scheme is found
+      if (!scheme) {
+        return next(request);
+      }
+      if (scheme.apiKeyLocation !== "header") {
```

Comment:
do we plan on supporting the other locations later? Why have them in the public surface if not?

Reply:
I believe that we're planning to support other locations later. Double checking with @timovv on this

Reply:
yeah I think we will be wanting to add support for others in future. But we could drop the others from the public surface for now since we don't support them yet

Reply:
If we don't have this union, the code gen will have to throw an error when generating, so I'd rather keeping the public surface for this union and aligns with TypeSpec.